### PR TITLE
Discard leftover input when consumed

### PR DESCRIPTION
**Pull request by gyscos**
The iterator given to `parse_event` now `take()` the value from `leftover`.

### CODE REVIEW

**Comment by gyscos** [comment]
Fixes #75

**Comment by gyscos** [comment]
New commit fixes the tests.
Since `parse_event` is now responsible to consume or not `leftover`, we don't need a special case for `\x1B` anymore.

File: src/input.rs [hunk 1]
```diff
@@ -32,6 +32,27 @@ pub struct Events<R> {
     leftover: Option<u8>,
 }
 
+/// An iterator that takes a single value from an option.
+struct OptionIterator<'a, T: 'a> {
```
Comment (gyscos) [inline src/input.rs]: I wished something like that were available in the std lib, but `std::option::Iter` is not quite what we want, as we can't easily keep the `Option` if `next()` is never called.

**Comment by gyscos** [comment]
Scratch what I said in the review, looks like it's possible to get by using `std::option::Iter`.

**Comment by ticki** [comment]
Sorry, I missed this PR again (dammit). LFTM.

**Pull request merged**